controllers/operands/deploymentHandler.go

```diff
@@ -15,23 +15,41 @@ import (
 	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
 )
 
+type deploymentOperand struct {
+	operand *genericOperand
+}
+
+func (dp deploymentOperand) ensure(req *common.HcoRequest) *EnsureResult {
+	return dp.operand.ensure(req)
+}
+
+func (dp deploymentOperand) reset() { /* no implementation */ }
+
 func newDeploymentHandler(Client client.Client, Scheme *runtime.Scheme, required *appsv1.Deployment) Operand {
-	h := &genericOperand{
-		Client: Client,
-		Scheme: Scheme,
-		crType: "Deployment",
-		hooks:  &deploymentHooks{required: required},
+	h := &deploymentOperand{
+		operand: &genericOperand{
+			Client: Client,
+			Scheme: Scheme,
+			crType: "Deployment",
+			hooks:  &deploymentHooks{required: required},
+		},
 	}
```

Comment:
We don't need the `h` variable here, because we don't use it nor modifying it later. Can just return it.
```suggestion
	return &deploymentOperand{
		operand: &genericOperand{
			Client: Client,
			Scheme: Scheme,
			crType: "Deployment",
			hooks:  &deploymentHooks{required: required},
		},
	}
```